lib/at_cmd_parser/src/at_params.c

```diff
@@ -0,0 +1,329 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <limits.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+#include <zephyr.h>
+#include <zephyr/types.h>
+#include <at_params.h>
+#include <kernel.h>
+
+#define AT_PARAMS_CALLOC k_calloc
+#define AT_PARAMS_MALLOC k_malloc
+#define AT_PARAMS_FREE k_free
+
+/* Internal function. Parameter cannot be null. */
```

Comment:
If the parameter cannot be `NULL`, then the function should either return an error or assert.
Do not rely on documentation to handle error conditions (here and other places).

Reply:
That is wrong. This is a static (private) function. The developer of the module is responsible to check the arguments before calling this internal function, and this is the case. We do not want to null check parameters of static function because this is not efficient and not needed if the module implementation is correct.